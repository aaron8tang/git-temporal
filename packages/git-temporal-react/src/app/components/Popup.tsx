import React from 'react';
import { style } from 'app/styles';

export interface PopupProps {
  isOpen: boolean;
  onClose: () => void;
  // The children are the popup content
  children: string | JSX.Element | JSX.Element[];
  style?: string | object;
  noBackdrop?: boolean;
}

const containerStyle = {
  overflow: 'visible',
};

const popupStyle = {
  _extends: 'popup',
  right: 0,
  marginRight: 0,
  minWidth: 180,
  zIndex: 4,
};

const backdropStyle = {
  opacity: 0,
  position: 'fixed',
  top: 0,
  left: 0,
  height: '100%',
  width: '100%',
  zIndex: 3,
};

const visible = { display: 'block' };
const hidden = { display: 'none' };

export class Popup extends React.Component<PopupProps> {
  render() {
    const popupStyles: any = [popupStyle, this.props.style];
    const backdropStyles: any = [backdropStyle];
    if (this.props.isOpen) {
      popupStyles.push(visible);
      backdropStyles.push(visible);
    } else {
      popupStyles.push(hidden);
      backdropStyles.push(hidden);
    }

    return (
      <div style={style(containerStyle)}>
        {!this.props.noBackdrop && (
          <div style={style(backdropStyles)} onClick={this.onClose} />
        )}
        <div style={style(popupStyles)} onClick={this.onClose}>
          {this.props.children}
        </div>
      </div>
    );
  }

  onClose = () => {
    this.props.onClose();
  };
}
