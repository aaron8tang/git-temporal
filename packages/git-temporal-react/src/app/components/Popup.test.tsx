import React from 'react';
import { shallow } from 'enzyme';

import { Popup } from './Popup';

describe('components/Popup', () => {
  describe('when rendered with isOpen=true)', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <Popup isOpen={true} onClose={jest.fn()}>
          This is some popup content, yo.
        </Popup>
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
