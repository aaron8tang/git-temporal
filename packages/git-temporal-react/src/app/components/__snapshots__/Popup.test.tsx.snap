// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Popup when rendered with isOpen=true) it should match snapshot 1`] = `
<div
  style={
    Object {
      "overflow": "visible",
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "display": "block",
        "height": "100%",
        "left": 0,
        "opacity": 0,
        "position": "fixed",
        "top": 0,
        "width": "100%",
        "zIndex": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "white",
        "borderRadius": 3,
        "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.2)",
        "display": "block",
        "marginRight": 0,
        "minWidth": 180,
        "padding": 10,
        "position": "absolute",
        "right": 0,
        "zIndex": 4,
      }
    }
  >
    This is some popup content, yo.
  </div>
</div>
`;
